packages/airnode-deployer/src/utils/infrastructure.ts

```diff
@@ -34,3 +35,91 @@ export class MultiMessageError extends Error {
     super(messages.join('\n'));
   }
 }
+
+export const BUCKET_NAME_REGEX = /^airnode-[a-f0-9]{12}$/;
+export const generateBucketName = () => `airnode-${randomBytes(6).toString('hex')}`;
+
+export enum FileSystemType {
+  Directory = 'Directory',
+  File = 'File',
+}
+
+export type FileSystemItem = Directory | File;
+
+export type Directory = {
+  type: FileSystemType.Directory;
+  bucketKey: string;
+  children: DirectoryStructure;
+};
+
+export type File = {
+  type: FileSystemType.File;
+  bucketKey: string;
+};
+
+export type DirectoryStructure = Record<string, FileSystemItem>;
+
+export const translatePathsToDirectoryStructure = (paths: String[]) => {
+  const directoryStructure: DirectoryStructure = {};
+
+  paths.forEach((path) => {
+    // Keep '/' at the end (for identifying directories)
+    path.split(/\/(?!$)/).reduce((ds, name: string, i: number, a: string[]) => {
```

Comment:
> Why /\/(?!$)/ regex

I added a better explanation why

> The reduce is pretty messy. I'd suggest just write a mutable for loop for this.

TBH, I don't think that would actually help the readability. It would have to be either as a recursion (which is basically what `reduce` is doing) or I would have to search for a correct spot in the directory structure for each path part which is also kinda counterintuitive.

I've added some more comments and changed the naming a bit. Let me know if it's better now.